Marlin/ultralcd.cpp

```diff
@@ -1013,17 +1022,23 @@ void lcd_cooldown() {
       current_position[Z_AXIS] = MESH_HOME_SEARCH_Z;
       plan_set_position(current_position[X_AXIS], current_position[Y_AXIS], current_position[Z_AXIS], current_position[E_AXIS]);
       lcd_goto_menu(_lcd_level_goto_next_point, true);
+      lcdDrawUpdate = LCDVIEW_CALL_REDRAW_NEXT;
     }
   }
 
   /**
    * Step 3: Display "Homing XYZ" - Wait for homing to finish
    */
   static void _lcd_level_bed_homing() {
-    if (lcdDrawUpdate) lcd_implementation_drawedit(PSTR(MSG_LEVEL_BED_HOMING), NULL);
-    if (axis_homed[X_AXIS] && axis_homed[Y_AXIS] && axis_homed[Z_AXIS])
+    lcd_implementation_drawedit(PSTR(MSG_LEVEL_BED_HOMING), NULL);
+    if (axis_homed[X_AXIS] && axis_homed[Y_AXIS] && axis_homed[Z_AXIS]) {
       lcd_goto_menu(_lcd_level_bed_homing_done);
-    lcdDrawUpdate = LCDVIEW_CALL_NO_REDRAW; // counts as a draw flag during graphical loop
+      lcdDrawUpdate = LCDVIEW_REDRAW_NOW;
+    }
```

Comment:
This might not be needed, because `lcd_goto_menu` sets `lcdDrawUpdate` to `LCDVIEW_CLEAR_CALL_REDRAW`.